File: 12. Rekursi/D. Kasur Rusak/d_kasur_rusak.go
package main

import (
	"fmt"
	"bufio"
	"os"
)

func palindrome(S string) bool{
	length := len([]rune(S))
	if length == 1{
		return true
	}else if length == 2{
		return S[0] == S[1]
	}else{
		return S[0] == S[length-1] && palindrome(S[1:length-1])
	}
}

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	scanner.Scan()
	if palindrome(scanner.Text()) {
		fmt.Println("YA")
	} else {
		fmt.Println("BUKAN")
	}
}

File: 12. Rekursi/D. Kasur Rusak/d_kasur_rusak.java
import java.util.Scanner;

public class d_kasur_rusak {
    public static boolean palindrome(String S) {
        int len = S.length();
        if (len == 1) {
            return true;
        } else if (len == 2) {
            return S.charAt(0) == S.charAt(1);
        } else {
            return S.charAt(0) == S.charAt(len - 1) && palindrome(S.substring(1, len - 1));
        }
    }

    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);
        String S = scan.nextLine();
        scan.close();
        if (palindrome(S)) {
            System.out.println("YA");
        } else {
            System.out.println("BUKAN");
        }
    }
}
